### PR TITLE
osd:load_pgs can be improved from single-threaded to multi-threaded

### PR DESCRIPTION
**Pull request by shangfufei**
Signed-off-by: shangfufei <shangfufei@inspur.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by shangfufei** [comment]
2019-04-09 10:03:19.883085 7f74f37c0a80  0  osd/OSD.cc:4570 osd.72 715 load_pgs
2019-04-09 10:25:34.539114 7f74f37c0a80  0  osd/OSD.cc:4702 osd.72 715 load_pgs opened 128 pgs

As shown in the log above, it takes more than 20 minutes in the load_pgs phase.

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -3948,6 +3948,7 @@ void OSD::load_pgs()
   }
 
   int num = 0;
+  //this place can be improve from single thread to multithread
```
Comment (xiexingguo) [inline src/osd/OSD.cc]: So it is really a proposal instead of a fix?

**Comment by xiexingguo** [comment]
Might want to promote the log level to further identify which are the truly time eat-up part(s) :-)

File: src/osd/OSD.cc (hunk 1 above)
Comment (shangfufei) [reply src/osd/OSD.cc]: > So it is really a proposal instead of a fix?

after optimization, the load_pg execution time can be controlled within 1 minute.

**Comment by shangfufei** [comment]
> Might want to promote the log level to further identify which are the truly time eat-up part(s) :-)
Log analysis shows that time is spent on the execution of read_log，just like follow log message:
2019-04-09 10:47:03.841978 7f55e2aaea80 20  osd/PGLog.cc:883 read_log 911'8079688 (911'8079687) modify   2:08cfa28f:::ffffffff03000661.00000000:head by mds.2.6787:5987866 2019-04-09 10:32:45.405754

**Comment by shangfufei** [comment]
what's wrong with  "make check failed", what can i do?

**Comment by shangfufei** [comment]
> Might want to promote the log level to further identify which are the truly time eat-up part(s) :-)

can you help me to review this code ?

**Comment by hjwsm1989** [comment]
@shangfufei  do you plan to open a new PR?

**Comment by shangfufei** [comment]
Sorry, the company asked that the code could not be submitted, and it will not be updated in the future.

**Pull request closed without merging**